src/modules/facility-covenant/facility-covenant.controller.ts

```diff
@@ -89,4 +91,35 @@ export class FacilityCovenantController {
   async getCovenantsForFacility(@Param() params: FacilityCovenantsParamsDto): Promise<GetFacilityCovenantsResponseDto[]> {
     return await this.facilityCovenantService.getCovenantsForFacility(params.facilityIdentifier);
   }
+
+  @Patch('/facilities/:facilityIdentifier/covenants')
+  @ApiOperation({
+    summary: 'Update all covenants for a facility.',
+  })
+  @ApiParam({
+    name: 'facilityIdentifier',
+    required: true,
+    type: 'string',
+    description: 'The identifier of the facility in ACBS.',
+    example: EXAMPLES.FACILITY_ID,
+  })
+  @ApiOkResponse({
+    description:
+      'The covenants for the facility have been updated. Due to limitations of ACBS, there is no 404 response; if the facility does not exist or it has no covenants, we will receive a 200 response',
```

Comment:
I've noticed missing full stop, which I'll add

Reply:
> Due to limitations of ACBS, there is no 404 response; if the facility does not exist or it has no covenants, we will receive a 200 response.

Do we always put this message on the description of the 200 response? It seems out of place, could we not put it in the descript of the ApiOperation instead?

Also, "we will receive" seems a bit unclear to me: how about "the endpoint will return"?

Reply:
GET facility covenants also mentions the lack of the 404 in the 200 response. I think the thing is that we want to make users aware that a 200 response does not guarantee that the facility/covenants exist. I propose keeping it here but changing it to `The covenants for the facility have been retrieved or the facility does not exist/has no covenants (due to limitations of ACBS, there is no 404 response).` What do you think?

Reply:
Ah I see, that's a good argument for having it here. Yes I like the updated message - can we do something similar to the GET description as well please?